library/network_connections.py

```diff
@@ -683,7 +683,21 @@ def connection_seems_active(cls, name):
             return None
 
         if i_content.strip() != "up":
-            return False
+            # For the active bonding/bridge connection, when all the port connection
+            # profiles are brought down firstly, then the operstate of the controller
+            # device will change into "down" automatically, which denotes missing the
+            # carrier for the controller device. However, the connection for the
+            # controller device stays active in such a situation.
```

Comment:
Thank you, @liangwen12year. The serialized test passed with your patch.
Let me just confirm this is the final solution. I somehow thought you were going to clean up the device (and connection?) completely in the bond tests. Your comment looks we allow the odd state instead... Is it ok?